### ISSUE TITLE
Doesn't support specification in JSON?

### ISSUE THREAD

**Issue by hbielenia** [Mermade/openapi-codegen#9]
OpenAPI specs can be either YAML or JSON. Loading JSON spec with openapi-codegen resulted in error:
```
hubiel@myhost ~/w/api-spec> node_modules/openapi-codegen/cg.js --verbose nodejs openapi.json 
Loaded config nodejs
Loaded definition openapi.json

/home/hubiel/work/api-spec/node_modules/js-yaml/lib/js-yaml/dumper.js:748
      throw new YAMLException('unacceptable kind of an object to dump ' + type);
      ^
YAMLException: unacceptable kind of an object to dump [object Undefined]
    at writeNode (/home/hubiel/work/api-spec/node_modules/js-yaml/lib/js-yaml/dumper.js:748:13)
    at writeBlockMapping (/home/hubiel/work/api-spec/node_modules/js-yaml/lib/js-yaml/dumper.js:627:10)
    at writeNode (/home/hubiel/work/api-spec/node_modules/js-yaml/lib/js-yaml/dumper.js:720:9)
    at dump (/home/hubiel/work/api-spec/node_modules/js-yaml/lib/js-yaml/dumper.js:809:7)
    at Object.safeDump (/home/hubiel/work/api-spec/node_modules/js-yaml/lib/js-yaml/dumper.js:815:10)
    at Object.transform (/home/hubiel/work/api-spec/node_modules/openapi-codegen/adaptor.js:683:32)
    at Object.main (/home/hubiel/work/api-spec/node_modules/openapi-codegen/index.js:30:13)
    at main (/home/hubiel/work/api-spec/node_modules/openapi-codegen/cg.js:155:19)
    at Object.<anonymous> (/home/hubiel/work/api-spec/node_modules/openapi-codegen/cg.js:197:4)
    at Module._compile (module.js:643:30)
```
The same definition was parsed without problems by swagger-node-codegen. It looks to me that openapi-codegen attempts to parse every input spec as YAML regardless of file extension or content.

**Comment by MikeRalphson** [comment]
>  It looks to me that openapi-codegen attempts to parse every input spec as YAML regardless of file extension or content.

JSON syntax is a subset of YAML so this should work without problem.

This is actually where we are setting the `swagger-yaml` property of the model (so it is writing yaml not reading it). It would be helpful to see your JSON definition if possible?

**Comment by hbielenia** [comment]
Here it is:
```
{
  "openapi": "3.0.0",
  "info": {
    "description": "REST API specification.",
    "version": "1.0.0",
    "title": "REST API"
  },
  "servers": [
    {
      "url": "{protocol}://{host}",
      "variables": {
        "protocol": {
          "enum": [
            "http",
            "https"
          ],
          "default": "http"
        },
        "host": {
          "enum": [
            "127.0.0.1:10010"
          ],
          "default": "127.0.0.1:10010"
        }
      }
    }
  ],
  "tags": [
    {
      "name": "sale",
      "description": "Sales reporting"
    }
  ],
  "paths": {
    "/sale": {
      "post": {
        "tags": [
          "sale"
        ],
        "summary": "Add new sale",
        "description": "Accepts new sale object to be recorded",
        "operationId": "addSale",
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/Sale"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ApiResponse"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "Sale": {
        "type": "object",
        "properties": {
          "click_ref": {
            "type": "string"
          },
          "currency": {
            "type": "string"
          },
          "order_id": {
            "type": "string"
          },
          "product_id": {
            "type": "string"
          },
          "turnover": {
            "type": "number",
            "multipleOf": 0.01
          },
          "quantity": {
            "type": "integer",
            "minimum": 1
          }
        },
        "required": [
          "click_ref",
          "order_id",
          "turnover",
          "quantity"
        ]
      },
      "ApiResponse": {
        "type": "object",
        "properties": {
          "success": {
            "type": "boolean"
          }
        },
        "required": [
          "success"
        ]
      }
    }
  }
}
```

**Comment by MikeRalphson** [comment]
Thanks! In this case it was the downconverter used to convert given OpenAPI 3.0 definitions into Swagger 2.0 for compatibility, it was creating `undefined` references in the converted definition.  The fix has been published as v1.3.1
